src/Propel/Runtime/Adapter/Pdo/PdoAdapter.php

```diff
@@ -451,27 +435,29 @@ public function createSelectSqlPart(Criteria $criteria, &$fromClause, $aliasAll
                 $parenPos = strrpos($columnName, '(');
                 $dotPos = strrpos($columnName, '.', ($parenPos !== false ? $parenPos : 0));
 
-                if ($dotPos !== false) {
-                    if ($parenPos === false) { // table.column
-                        $tableName = substr($columnName, 0, $dotPos);
-                    } else { // FUNC(table.column)
-                        // functions may contain qualifiers so only take the last
-                        // word as the table name.
-                        // COUNT(DISTINCT books.price)
-                        $tableName = substr($columnName, $parenPos + 1, $dotPos - ($parenPos + 1));
-                        $lastSpace = strrpos($tableName, ' ');
-                        if ($lastSpace !== false) { // COUNT(DISTINCT books.price)
-                            $tableName = substr($tableName, $lastSpace + 1);
-                        }
-                    }
-                    // is it a table alias?
-                    $tableName2 = $criteria->getTableForAlias($tableName);
-                    if ($tableName2 !== null) {
-                        $fromClause[] = $tableName2 . ' ' . $tableName;
-                    } else {
-                        $fromClause[] = $tableName;
+                if ($dotPos === false) {
+                    continue;
+                }
+
+                if ($parenPos === false) { // table.column
+                    $tableName = substr($columnName, 0, $dotPos);
+                } else { // FUNC(table.column)
+                    // functions may contain qualifiers so only take the last
+                    // word as the table name.
+                    // COUNT(DISTINCT books.price)
+                    $tableName = substr($columnName, $parenPos + 1, $dotPos - ($parenPos + 1));
+                    $lastSpace = strrpos($tableName, ' ');
+                    if ($lastSpace !== false) { // COUNT(DISTINCT books.price)
+                        $tableName = substr($tableName, $lastSpace + 1);
                     }
                 }
+                // is it a table alias?
+                $tableName2 = $criteria->getTableForAlias($tableName);
+                if ($tableName2 !== null) {
```

Comment:
I can see that this is a copy paste, but personally I don't like `else` statements :)

if ($tableName2 !== null) {
$tableName2 = $tableName2 . '  ';
}
$fromClause[] = (string)$tableName2 . $tableName;

But this is not a mandatory to fix